azure/scope/managedcontrolplane_test.go

```diff
@@ -1453,3 +1453,95 @@ func TestManagedControlPlaneScope_PrivateEndpointSpecs(t *testing.T) {
 		})
 	}
 }
+
+func TestManagedControlPlaneScope_AutoUpgradeProfile(t *testing.T) {
+	cases := []struct {
+		Name     string
```

Comment:
```suggestion
		name     string
		input    ManagedControlPlaneScopeParams
		expected *managedclusters.ManagedClusterAutoUpgradeProfile
```

Nit: these don't need to be exported